src/libstd/sys/unix/fs.rs

```diff
@@ -732,17 +791,126 @@ pub fn canonicalize(p: &Path) -> io::Result<PathBuf> {
 }
 
 pub fn copy(from: &Path, to: &Path) -> io::Result<u64> {
-    use fs::{File, set_permissions};
-    if !from.is_file() {
+    use fs::OpenOptions;
+    let mut from_opts = OpenOptions::new();
+    let mut reader = try!(from_opts.read(true).open(&from));
```

Comment:
Now I don't remember why I changed that...
If I don't find the reason I will change it back.